server/parameters/CoreSubcommandParser.java

```diff
@@ -34,34 +34,35 @@
 import static com.vaticle.typedb.core.common.exception.ErrorMessage.Server.CLI_OPTION_UNRECOGNISED;
 import static com.vaticle.typedb.core.common.iterator.Iterators.iterate;
 
-public class ServerSubcommandParser {
+public class CoreSubcommandParser {
 
-    private static void validateRequiredOptions(Set<OptionParser> requiredParsers, Set<Option> options) {
+    protected static void validateRequiredOptions(Set<OptionParser> requiredParsers, Set<Option> options) {
         requiredParsers.forEach(parser -> {
             if (iterate(options).noneMatch(option -> option.name().equals(parser.name()))) {
                 throw TypeDBException.of(CLI_OPTION_REQUIRED, parser.name());
             }
         });
     }
 
-    private static void validateUnrecognisedOptions(Set<OptionParser> recognisedParsers, Set<Option> options) {
-        recognisedParsers.forEach(parser -> {
-            if (iterate(options).noneMatch(option -> option.name().equals(parser.name()))) {
-                throw TypeDBException.of(CLI_OPTION_UNRECOGNISED, parser);
+    protected static void validateUnrecognisedOptions(Set<OptionParser> recognisedParsers, Set<Option> options) {
+        options.forEach(option -> {
```

Comment:
this was a bug - we throwing `UNRECOGNISED` whenever there was a parser that was not in the options, rather than the other way around.